ocaml/parsing/parser.mly

```diff
@@ -127,7 +127,8 @@ let neg_string f =
   then String.sub f 1 (String.length f - 1)
   else "-" ^ f
 
-let mkuminus ~oploc name arg =
+let mkuminus ~loc ~oploc name arg =
```

Comment:
I believe this duplicates the attributes in the non-constant case. (Same for `mkuplus`.)

Reply:
Why do you believe that? I don't see it. This function is essentially decomposing the input `arg` and using only its `desc`. My change also preserves its attributes.

Reply:
The full implementation is:

```ocaml
let mkuminus ~loc ~oploc name arg =
  mkexp ~loc ~attrs:arg.pexp_attributes @@
  match name, arg.pexp_desc with
  | "-", Pexp_constant(Pconst_integer (n,m)) ->
      Pexp_constant(Pconst_integer(neg_string n,m))
  | ("-" | "-."), Pexp_constant(Pconst_float (f, m)) ->
      Pexp_constant(Pconst_float(neg_string f, m))
  | _ ->
      Pexp_apply(mkoperator ~loc:oploc ("~" ^ name), [Nolabel, arg])
```

Note how the `mkexp ~loc ~attrs:arg.pexp_attributes` can be applied to a `Pexp_apply` node that has `arg` as a subtree.

Reply:
Oh -- I missed that. You're right. Will fix shortly.